tsl/test/expected/vector_agg_param.out

```diff
@@ -47,4 +60,75 @@ select * from unnest(array[0, 1, 2]::int[]) x, lateral (select sum(a) from pvagg
  2 | 1498500
 (3 rows)
 
+explain (verbose, costs off)
+select * from unnest(array[0, 1, 2]::int[]) x, lateral (select sum(a + x) from pvagg) xx;
+                                                                                                        QUERY PLAN                                                                                                         
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ Nested Loop
+   Output: x.x, (sum((_hyper_1_1_chunk.a + x.x)))
+   ->  Function Scan on pg_catalog.unnest x
+         Output: x.x
+         Function Call: unnest('{0,1,2}'::integer[])
+   ->  Finalize Aggregate
+         Output: sum((_hyper_1_1_chunk.a + x.x))
+         ->  Append
+               ->  Partial Aggregate
+                     Output: PARTIAL sum((_hyper_1_1_chunk.a + x.x))
+                     ->  Custom Scan (DecompressChunk) on _timescaledb_internal._hyper_1_1_chunk
+                           Output: _hyper_1_1_chunk.a
+                           ->  Seq Scan on _timescaledb_internal.compress_hyper_2_3_chunk
+                                 Output: compress_hyper_2_3_chunk._ts_meta_count, compress_hyper_2_3_chunk.s, compress_hyper_2_3_chunk._ts_meta_min_1, compress_hyper_2_3_chunk._ts_meta_max_1, compress_hyper_2_3_chunk.a
+               ->  Partial Aggregate
+                     Output: PARTIAL sum((_hyper_1_2_chunk.a + x.x))
+                     ->  Custom Scan (DecompressChunk) on _timescaledb_internal._hyper_1_2_chunk
+                           Output: _hyper_1_2_chunk.a
+                           ->  Seq Scan on _timescaledb_internal.compress_hyper_2_4_chunk
+                                 Output: compress_hyper_2_4_chunk._ts_meta_count, compress_hyper_2_4_chunk.s, compress_hyper_2_4_chunk._ts_meta_min_1, compress_hyper_2_4_chunk._ts_meta_max_1, compress_hyper_2_4_chunk.a
+(20 rows)
+
+select * from unnest(array[0, 1, 2]::int[]) x, lateral (select sum(a + x) from pvagg) xx;
```

Comment:
Not sure what this is testing. Is it just checking that the query doesn't fail (if it did fail prior to this fix)? Or is it testing that it gives correct output?

How can I know that the output (sum) is correct? Is there a non-compressed (regular) table I can compare with?

Reply:
This is testing an aggregate function reference that has an expression that references a nested loop parameter.

I generated a reference by running the same query with vectorized aggregation disabled.

Reply:
Shouldn't the reference output be part of the test? Now the `set` for turning off vector agg is a  commented line so I cannot see the reference or verify that this is correct.

Reply:
You uncomment this line and this generates the reference in the output file -- all queries run with normal postgres aggregation and not vectorized aggregation. Then you comment it back and run the test again and check that nothing else changes. You do this once when changing the test, I already done this, the test output has the correct results generated by standard Postgres plan. This is the approach I use for some other tests as well.

Reply:
The pattern we have for this is to generate two output files and do a diff between them in the test. There are examples in other tests how to do this.

Reply:
> The pattern we have for this is to generate two output files and do a diff between them in the test. There are examples in other tests how to do this.

Yeah, I know, I don't like to use it because:

1. The test runs more than twice slower
1. Inconvenient to view the entire test reference, you have to do extra steps to open another file for this.
1. The PG version is not always the same as our version, e.g. some vectorized functions have better numeric stability.

This is the stuff that I remember off the top of my head, probably there are more reasons. Why is it a problem?

Reply:
Oh, you also have to put the actual test queries into a separate file and run it with psql, so editing a test is also more complicated.

Reply:
> Having this comparison of the outputs is good because it also easily captures future errors and regressions.

The test I wrote also compares the outputs, only the PG output is fixed at the test editing time.

When you generate the output each time, you make it effectively compare the four different supported PG versions against each other. Not sure what's the benefit, probably you'll just run into some numeric stability change in PG and will have to painfully work around it.

Reply:
> > The pattern we have for this is to generate two output files and do a diff between them in the test. There are examples in other tests how to do this.
> 
> Yeah, I know, I don't like to use it because:
> 
>     1. The test runs more than twice slower
> 
>     2. Inconvenient to view the entire test reference, you have to do extra steps to open another file for this.
> 
>     3. The PG version is not always the same as our version, e.g. some vectorized functions have better numeric stability.
> 
> 
> This is the stuff that I remember off the top of my head, probably there are more reasons. Why is it a problem?

I  am merely giving feedback on things I think would improve the test and avoid regression, as well as for my own understanding so that I don't just approve without understanding what is going on. Only now, after I asked, it is clear that the test output is in fact different from regular PG aggregates, as you admit. Even if it is not strictly wrong,  I cannot verify this in the review. It gives me pause because it was neither documented nor clear from the test. At the very least, this could have been good information to provide in the test. Having different aggregate output also means we can't easily capture regressions and it requires someone to know that they need to manually enable and inspect the output when something changes, which I think you are currently the only person who knows how to do easily.

Ideally, our tests should be easy to understand and maintain also by others, this is the perspective I have. Is there some way we can improve the test to make it easier for others to understand the aspects above?

Reply:
> Only now, after I asked, it is clear that the test output is in fact different from regular PG aggregates, as you admit.

That's not in this test, that's in different ones where I also use this pattern. What regressions do you want to avoid? This is the usual "golden test", it runs some queries and compares their output against the one captured in the reference. Most our tests are like that. Here we also have a possibility to compare the reference against the analogous PG output by uncommenting a single line in this test. What should be improved here?